### PR TITLE
add get salaries and cities names interactor and tests

### PR DESCRIPTION
**Pull request by HaidyAbuGom3a**

### CODE REVIEW

File: src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+class GetSalaryAverageAndCitiesNamesInCountryInteractor(
+    private val dataSource: CostOfLivingDataSource
+) {
+    fun execute(countryName:String):List<Pair<String,Float>>{
+        return if (checkIfCountryIsValid(countryName))
+            getAllCitiesWithLowercaseCountryName()
+            .filter { it.country == countryName.lowercase() }
+            .filter{excludeNullSalariesAndLowQualityData(it)}
+            .map{Pair(it.cityName,it.averageMonthlyNetSalaryAfterTax!!) }
```
Comment (flutterdevtaha) [inline src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt:14]: if you like, there is a more consice syntax in this one in filter

File: src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+    private fun checkIfCountryIsValid(countryName: String):Boolean{
+        return countryName.lowercase() in getAllCitiesWithLowercaseCountryName()
+            .map { it.country }
+    }
+
+    private fun getAllCitiesWithLowercaseCountryName():List<CityEntity>{
+        return dataSource.getAllCitiesData()
+                .map{convertCountryNameToLower(it)}
```
Comment (flutterdevtaha) [inline src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt:27]: last note

File: src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+    private fun convertCountryNameToLower(city:CityEntity):CityEntity{
```
Comment (flutterdevtaha) [inline src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt:29]: there is no need for this whole function, instead do what it need for the country only

**Review by flutterdevtaha — CHANGES_REQUESTED** [review]

File: src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        return CityEntity(
```
Comment (ma28alhams) [inline src/main/kotlin/interactor/GetSalaryAverageAndCitiesNamesInCountryInteractor.kt:30]: Copy the object then change the related data

**Pull request closed without merging**